src/threading/SerialTypes.h

```diff
@@ -126,9 +126,8 @@ struct Field {
  * those Vals supported).
  */
 struct Value {
-    TypeTag type;         //! The type of the value.
-    TypeTag subtype;      //! Inner type for sets and vectors.
-    bool present = false; //! False for optional record fields that are not set.
+    TypeTag type;    //! The type of the value.
+    TypeTag subtype; //! Inner type for sets and vectors.
```

Comment:
TypeTag are both 4 bytes each due to being ints by dfeault, right? I'm not sure about the implications, but the following might save 6 bytes (minus padding - might make sense to move present up if that's done). We may have a few thousand threading::Vals flying around, so that might  be worth it.

```
diff --git a/src/Type.h b/src/Type.h
index e7dc3c52a2..fbd5b5b0b0 100644
--- a/src/Type.h
+++ b/src/Type.h
@@ -59,7 +59,7 @@ public:
 } // namespace detail
 
 // Zeek types.
-enum TypeTag {
+enum TypeTag : uint8_t {
```

Reply:
`InternalTypeTag` being a `uint16_t` is strange also, since it's only 9 values. Let me poke at that.

Reply:
This one ends up changing a lot of other objects too, so I'm gonna make a note of it and leave it out for now. There's definitely space savings we could make across all of the `enum`s we have in the code. I'll take a look at it more deeply the next time I do this.